### PR TITLE
New data source `azurerm_communication_service`

### LINKED ISSUE #21736: Support for azurerm_communication_services data object
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

A data object to read config from a existing Azure communication service could be helpful in scenarios where the ACS is created somewhere not managed by Terraform. This will allow for a simple way to, for example, read the connection string to be put into a keyvault for our other systems to access. 

As a workaround we are now using a null_object to read the connection string with a command, and outputting that to a file on our build environment.

### New or Affected Resource(s)/Data Source(s)

azurerm_communication_service

### Potential Terraform Configuration

```hcl
data "azurerm_communication_service" "example" {
  name = "existing"
  resource_group_name = "example"
}

output "connection_string" {
  value = data.azurerm_communication_service.example.primary_connection_string
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by myc2h6o**
Close #21736

### CODE REVIEW

**Comment by myc2h6o** [comment]
test result
image

File: internal/services/communication/communication_service_data_source.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+				if response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("%s was not found", id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			keysResp, err := client.ListKeys(ctx, id)
+			if err != nil {
+				return fmt.Errorf("listing keys for %s: %+v", id, err)
```
Comment (tombuildsstuff) [inline internal/services/communication/communication_service_data_source.go:110]: data sources should gracefully handle having limited permissions, so we shouldn't raise this error when there's a permissions error (presumably a 401?)

File: internal/services/communication/communication_service_data_source.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+					state.PrimaryKey = *model.PrimaryKey
+				}
+
+				if model.SecondaryConnectionString != nil {
+					state.SecondaryConnectionString = *model.SecondaryConnectionString
+				}
+
+				if model.SecondaryKey != nil {
+					state.SecondaryKey = *model.SecondaryKey
+				}
```
Comment (tombuildsstuff) [inline internal/services/communication/communication_service_data_source.go:138]: FWIW we can reduce this to:

```suggestion
				state.PrimaryConnectionString = pointer.From(model.PrimaryConnectionString)
				state.PrimaryKey = pointer.From(model.PrimaryKey)
				state.SecondaryConnectionString = pointer.From(model.SecondaryConnectionString)
				state.SecondaryKey = pointer.From(model.SecondaryKey)
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @myc2h6o 

Thanks for this PR - I've taken a look through and left some comments inline, but if we can fix those up then this should otherwise be good to go 👍 

Thanks!

**Comment by myc2h6o** [comment]
@tombuildsstuff thanks for reviewing! I've updated the code to use `pointer.From`, and used a warning message to replace the error. Please take a look.

**Comment by myc2h6o** [comment]
updated test result
image

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for pushing those changes @myc2h6o

**Pull request merged**